### DISCUSSION TITLE
GitHub Actions workflow changes

### DISCUSSION THREAD

**Discussion by codefiles** [archlinux/archinstall#1748] [Ideas]
The following are my proposed changes to the GitHub Actions workflow.

- Package archinstall.
  - Additional workflow called `package.yml`.
  - This workflow would build an Arch Linux package `archinstall-git` using the commit that triggered it.
  - This package would then be uploaded as an artifact.
- Change `iso-build.yaml` to install archinstall as a package.
  - Pull in `archinstall-git` package from artifact.
  - Create a local repo (`repo-add`) and add `archinstall-git` package.
  - Configure Archiso `pacman.conf` to use the local repo.
  - Replace `archinstall` with `archinstall-git` in `packages.x86_64`.
  
This would have the added benefits of testing packaging, making the package available for download, and would better simulate the installation ISO environment for testing. The process outlined in the changes to `iso-build.yaml` are derived from the ArchWiki. [1]

The current process being used to build the ISO with the commit that triggered it has some weird side effects. For example opening `tmux` while in the ISO environment will re-trigger the archinstall installation process as will opening additional sessions.

I wanted to bring this up for discussion because I am unsure if this change would be welcomed. It is possible the current method for installing archinstall into the testing ISO is more desirable for reasons that I am unaware of.

I build my own Archiso with archinstall installed as the package `archinstall-git` for testing and it works fine. There is no issue in working with the project files either since they can be found at the path `/usr/lib/python3.10/site-packages/archinstall/`.

@Torxed if you like the proposal I can put together a pull request.

[1] https://wiki.archlinux.org/title/Archiso#Custom_local_repository

**Comment by svartkanin** [comment]
I think reworking the workflows is a good idea, they aren't really doing what they should be doing right now I think.
I started updating the existing ones in my PR to at least cover the basics but it definitely needs more work and a separate PR would be better

**Reply by codefiles** [reply]
I like that you moved the commands to build the ISO from `iso-build.yaml` to `build_iso.sh`. Keep in mind that it is making it difficult to contribute to this project with the ever expanding size of your PyParted pull request which appears to me as it will be merged since it contains critical improvements.

It seems your comment only addresses the discussion title and makes no acknowledgement of the discussion body.



> I think reworking the workflows is a good idea, they aren't really doing what they should be doing right now I think.

Could you expand on this? What changes to the workflows would have them "doing what they should be doing"?

**Reply by svartkanin** [reply]
Agreed, I'd be very thrilled on getting it merged and then everyone can do changes based on it rather than working off master and most contributors may not rven know about the large change coming.

I updated the workflows in the PR as they don't detect any dependency failures or issues atm which you had already found during testing as well. Installing archinstall with the listed steps there doesn't actually install dependencies.

- Building the arch iso is a good step, but it doesn't actually test if archinstall is running fine

- building the package is done but running archinstall is jot which could fail

- when installing the packages verifying that the root structure is as expected

**Reply by codefiles** [reply]
Right, the `package.yml` workflow would be just to test that the package can successfully be built from the triggering commit. Again, with it we could also have an artifact that contains the built package which is useful in itself but could also be used for the ISO build workflow.

I agree that the workflows could use the additional tests that you proposed. I see them as separate from a workflow for building the package but you bring up a great point which is the build of the package can succeed when dependencies are missing.
